resin/main.go

```diff
@@ -217,5 +140,25 @@ func main() {
 		sshConfig.KeyboardInteractiveCallback = auth.keyboardInteractiveCallback
 	}
 
-	sshproxy.New(viper.GetString("dir"), viper.GetString("shell"), viper.GetBool("allow-env"), sshConfig).Listen(viper.GetString("port"))
+	if viper.GetString("sentry-dsn") != "" {
+		if err := raven.SetDSN(viper.GetString("sentry-dsn")); err != nil {
+			log.Fatal("Sentry initialisation failed", err)
+		}
+	}
+
+	server, err := sshproxy.New(
+		viper.GetString("dir"),
+		viper.GetString("shell"),
+		viper.GetBool("allow-env"),
+		sshConfig,
+		func(err error, tags map[string]string) {
+			log.Printf("ERROR: %s", err)
+			raven.CaptureErrorAndWait(err, tags)
```

Comment:
* good point, I was thinking differently when I initially used that but you're right `CaptureError` makes more sense
* yes :)